### PR TITLE
Fix namefilter in pipelinerun list page causing memory leaks

### PR DESCRIPTION
**Pull request by karthikjeeyar**
## Fixes 

https://issues.redhat.com/browse/RHTAPBUGS-578
https://issues.redhat.com/browse/RHTAPBUGS-584 

## Description

when a user tries to use the name filter the browser goes unresponsive.

## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

https://github.com/openshift/hac-dev/assets/9964343/540297c1-463b-48d1-a5fa-f5911d9f38f0

https://github.com/openshift/hac-dev/assets/9964343/aa8bd028-1073-45ce-a8a2-89c04b3852a6

Commit pipelineruns tab:

image

## Browser conformance: 

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: src/components/PipelineRunListView/PipelineRunsListView.tsx [hunk 1]
```diff
@@ -33,8 +34,18 @@ const PipelineRunsListView: React.FC<PipelineRunsListViewProps> = ({ application
   const { namespace } = useWorkspaceInfo();
   const [components, componentsLoaded] = useComponents(namespace, applicationName);
   const [nameFilter, setNameFilter] = useSearchParam('name', '');
+  const [name, setName] = React.useState('');
   const [statusFilterExpanded, setStatusFilterExpanded] = React.useState<boolean>(false);
   const [statusFiltersParam, setStatusFiltersParam] = useSearchParam('status', '');
+  const requestQueue = React.useRef<Function[]>([]);
+  const [onLoadName, setLoadName] = React.useState(nameFilter);
```
Comment (abhinandan13jan) [inline src/components/PipelineRunListView/PipelineRunsListView.tsx:41]: setOnloadName?

**Review by rottencandy — COMMENTED** [review]
I noticed that the vulnerabilities column seems to be stuck in a loading state and only completes once a row is filtered through.

https://github.com/openshift/hac-dev/assets/20013884/320a3ef2-2425-4390-b407-8f45de8b3f4a

**Comment by karthikjeeyar** [comment]
@rottencandy I have fixed this issue, please re-test

**Review by rottencandy — APPROVED** [review]
/lgtm

**Pull request merged**